packages/token-bridge-sdk/src/util/deposits.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  L1TransactionReceipt,
+  L1ToL2MessageStatus,
+  EthDepositStatus
+} from '@arbitrum/sdk'
+import { Provider } from '@ethersproject/providers'
+import { AssetType } from '../hooks/arbTokenBridge.types'
+import { Transaction } from '../hooks/useTransactions'
+import { getL1TokenData } from '../util'
+
+export const updateAdditionalDepositData = async (
+  depositTx: Transaction,
+  l1Provider: Provider,
+  l2Provider: Provider
+) => {
+  // 1. for all the fetched txns, fetch the transaction receipts and update their exact status
+  // 2. on the basis of those, finally calculate the status of the transaction
+
+  // fetch L1 transaction receipt
+  const depositTxReceipt = await l1Provider.getTransactionReceipt(
+    depositTx.txID
+  )
+  const l1TxReceipt = new L1TransactionReceipt(depositTxReceipt)
+
+  // Check if deposit is ETH
+  if (depositTx.assetName === AssetType.ETH) {
+    // from the receipt - get the eth-deposit-message
+    const [ethDepositMessage] = await l1TxReceipt.getEthDeposits(l2Provider)
+
+    if (!ethDepositMessage) {
+      return
+    }
+
+    // from the eth-deposit-message, extract more things like retryableCreationTxID, status, etc
+    const status = await ethDepositMessage.status()
+    const isDeposited = status === EthDepositStatus.DEPOSITED
+    const timestampCreated = depositTx.blockNumber
+      ? (await l1Provider.getBlock(Number(depositTx.blockNumber))).timestamp *
```

Comment:
The cast to `Number` here doesn't seem to be necessary

Reply:
Done